### PR TITLE
Inventory generation fixes

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary:
- Updated InventoryFilters package to incorporate layout changes
- Dropped ordered_yaml filter
- Updated data parser to include parse yaml into ordered dict
- Drop camel library dependency
- Added _convert_layout funtion in API to change the structure of layout

Fixes: #540

### CODE REVIEW

**Comment by herlo** [comment]
[test]

**Comment by dyasny** [comment]
LGTM, tested 5 times with no glitches

File: linchpin/cli/__init__.py [hunk 1]
```diff
@@ -6,7 +6,6 @@
 import sys
 import json
 import click
-
```
Comment (herlo) [inline linchpin/cli/__init__.py]: Style nit. When doing imports vs from statements, they are separated by an empty line.

**Comment by samvarankashyap** [comment]
@p3ck @herlo Docker build failed ! ??

**Comment by herlo** [comment]
@samvarankashyap I'm rebuilding this. It failed because #587 is rebuilding containers but isn't merged yet.

**Comment by herlo** [comment]
@samvarankashyap @p3ck https://jenkins-continuous-infra.apps.ci.centos.org/job/ci-linchpin/27/

**Comment by herlo** [comment]
[test]

**Pull request merged**